docs/ql-documentation/learn-ql/python/pointsto-type-infer.rst

```diff
@@ -186,38 +191,38 @@ The original query looked this:
-   select call, "call to 'input'."
+   where call.getFunc() = name and name.getId() = "eval"
+   select call, "call to 'eval'."
 
-➤ `See this in the query console <https://lgtm.com/query/690010037/>`__. Two of the demo projects on LGTM.com have calls that match this pattern.
+➤ `See this in the query console <https://lgtm.com/query/6718356557331218618/>`__. Two of the demo projects on LGTM.com have calls that match this pattern.
 
 There are two problems with this query:
 
--  It assumes that any call to something named "input" is a call to the builtin ``input`` function, which may result in some false positive results.
--  It assumes that ``input`` cannot be referred to by any other name, which may result in some false negative results.
+-  It assumes that any call to something named "eval" is a call to the builtin ``eval`` function, which may result in some false positive results.
+-  It assumes that ``eval`` cannot be referred to by any other name, which may result in some false negative results.
 
-We can get much more accurate results using call-graph analysis. First, we can precisely identify the ``FunctionObject`` for the ``input`` function, by using the ``builtin_object`` QL predicate as follows:
+We can get much more accurate results using call-graph analysis. First, we can precisely identify the ``FunctionValue`` for the ``eval`` function, by using the ``Value::named`` QL predicate as follows:
 
 .. code-block:: ql
 
    import python
 
-   from FunctionObject input
-   where input = builtin_object("input")
-   select input
+   from Value eval
+   where eval = Value::named("eval")
+   select eval
 
-Then we can use ``FunctionObject.getACall()`` to identify calls to the ``input`` function, as follows:
+Then we can use ``Value.getACall()`` to identify calls to the ``eval`` function, as follows:
 
 .. code-block:: ql
 
    import python
 
-   from ControlFlowNode call, FunctionObject input
-   where input = builtin_object("input") and
-         call = input.getACall()
-   select call, "call to 'input'."
+   from ControlFlowNode call, Value eval
+   where eval = Value::named("eval") and
+         call = eval.getACall()
+   select call, "call to 'eval'."
 
-➤ `See this in the query console <https://lgtm.com/query/670490037/>`__. This accurately identifies calls to the builtin ``input`` function even when they are referred to using an alternative name. Any false positive results with calls to other ``input`` functions, reported by the original query, have been eliminated. It finds one result in files referenced by the *saltstack/salt* project.
+➤ `See this in the query console <https://lgtm.com/query/535131812579637425/>`__. This accurately identifies calls to the builtin ``eval`` function even when they are referred to using an alternative name. Any false positive results with calls to other ``eval`` functions, reported by the original query, have been eliminated. It finds one result in files referenced by the *saltstack/salt* project.
```

Comment:
This seems to generate more results than it used to, so the text should probably be revised.